File: DemoObjC/DemoObjC/View Controllers/In-App/RewardedAdVC.m
//
//  RewardedAdVC.m
//  DemoObjC
//
// © 2026 Limelight Inc. All Rights Reserved.
//

#import "RewardedAdVC.h"
@import LimelightSDK;

// MARK: - Private Interface

@interface RewardedAdVC () <LLRewardedAdDelegate>

// MARK: Storyboard UI

@property(weak, nonatomic) IBOutlet UIButton *showAdButton;
@property(weak, nonatomic) IBOutlet UILabel *adUnitIdLabel;
@property(weak, nonatomic) IBOutlet UILabel *adSizeLabel;

// InterstitialAdDelegate
@property(weak, nonatomic) IBOutlet UILabel *onAdLoadedLabel;
@property(weak, nonatomic) IBOutlet UILabel *onAdFailedToLoadLabel;
@property(weak, nonatomic) IBOutlet UILabel *onAdOpenedLabel;
@property(weak, nonatomic) IBOutlet UILabel *onAdImpressionLabel;
@property(weak, nonatomic) IBOutlet UILabel *onAdClickedLabel;
@property(weak, nonatomic) IBOutlet UILabel *onAdClosedLabel;
@property(weak, nonatomic) IBOutlet UILabel *onAdRewardedLabel;

// Controls
@property(weak, nonatomic) IBOutlet UIButton *refreshButton;

- (IBAction)didClickShowAdButton:(id)sender;
- (IBAction)didClickRefreshButton:(id)sender;

// MARK: Ad View

@property (strong, nonatomic) LLRewardedAd *rewardedAd;

@end

// MARK: - Implementation

@implementation RewardedAdVC

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setupAndLoadAd];
}

- (void)setupAndLoadAd {
    self.adUnitIdLabel.text = self.adConfigId;
    self.adSizeLabel.text = [NSString stringWithFormat:@"%dx%d", (int)self.adSize.width, (int)self.adSize.height];
    
    NSSet<LLAdFormat *> *adFormats = [NSSet setWithObjects:LLAdFormat.banner, LLAdFormat.video, nil];
    
    self.rewardedAd = [[LLRewardedAd alloc] initWithAdUnitId:self.adConfigId];
    self.rewardedAd.delegate = self;
    self.rewardedAd.supportSKOverlay = NO;
    self.rewardedAd.adSize = self.adSize;
    self.rewardedAd.adFormats = adFormats;
    [self.rewardedAd loadAd];
}

- (IBAction)didClickShowAdButton:(id)sender {
    [self showAd];
}

- (void)showAd {
    [self.rewardedAd showFrom:self];
}

- (void)resetEvents {
    self.onAdLoadedLabel.enabled = NO;
    self.onAdFailedToLoadLabel.enabled = NO;
    self.onAdOpenedLabel.enabled = NO;
    self.onAdImpressionLabel.enabled = NO;
    self.onAdClickedLabel.enabled = NO;
    self.onAdClosedLabel.enabled = NO;
    self.onAdRewardedLabel.enabled = NO;
}

- (IBAction)didClickRefreshButton:(id)sender {
    [self refreshAd];
}

- (void)refreshAd {
    [self resetEvents];
    self.refreshButton.enabled = NO;
    [self.rewardedAd loadAd];
}

@end

// MARK: - LimelightRewardedAdDelegate

@implementation RewardedAdVC (LLRewardedAdDelegate)

- (void)onAdLoaded:(LLRewardedAd *)rewardedAd :(LLAd * _Nonnull)ad {
    [self resetEvents];
    self.onAdLoadedLabel.enabled = YES;
    self.showAdButton.enabled = YES;
    self.refreshButton.enabled = YES;
}

- (void)onAdFailedToLoad:(LLRewardedAd *)rewardedAd error:(LLError *)error {
    [self resetEvents];
    self.onAdFailedToLoadLabel.enabled = YES;
    self.refreshButton.enabled = YES;
}

- (void)onAdOpened:(LLRewardedAd *)rewardedAd {
    self.onAdOpenedLabel.enabled = YES;
}

- (void)onAdImpression:(LLRewardedAd *)rewardedAd {
    self.onAdImpressionLabel.enabled = YES;
}

- (void)onAdClicked:(LLRewardedAd *)rewardedAd {
    self.onAdClickedLabel.enabled = YES;
}

- (void)onAdClosed:(LLRewardedAd *)rewardedAd {
    self.onAdClosedLabel.enabled = YES;
}

- (void)onAdRewarded:(LLRewardedAd *)rewardedAd reward:(LLReward *)reward {
    self.onAdRewardedLabel.enabled = YES;
}

@end

File: DemoSwift/DemoSwift/View Controllers/In-App/RewardedAdVC.swift
//
//  RewardedAdVC.swift
//  DemoSwift
//
// © 2026 Limelight Inc. All Rights Reserved.
//

import UIKit
import LimelightSDK

// MARK: - RewardedAdVC
final class RewardedAdVC: UIViewController {
    
    // MARK: Storyboard UI
    
    @IBOutlet var showAdButton: UIButton!
    @IBAction func didClickShowAdButton(_ sender: Any) { showAd() }
    // Ad Info
    @IBOutlet var adUnitIdLabel: UILabel!
    @IBOutlet var adSizeLabel: UILabel!
    // InterstitialAdDelegate
    @IBOutlet var onAdLoadedLabel: UILabel!
    @IBOutlet var onAdFailedToLoadLabel: UILabel!
    @IBOutlet var onAdOpenedLabel: UILabel!
    @IBOutlet var onAdImpressionLabel: UILabel!
    @IBOutlet var onAdClickedLabel: UILabel!
    @IBOutlet var onAdClosedLabel: UILabel!
    @IBOutlet var onAdRewardedLabel: UILabel!
    // Controls
    @IBOutlet var refreshButton: UIButton!
    @IBAction func didClickRefreshButton(_ sender: Any) { refreshAd() }

    // MARK: Ad View
    
    var rewardedAd: LimelightRewardedAd?
    
    // Ad View Customizations
    var adConfigId = "1234567890"
    var adSize: CGSize = .init(width: 480, height: 320)
    var adFormats: Set<LimelightAdFormat> = [.banner, .video]
    // SKAdNetwork
    var supportSKOverlay = false
    
    // MARK: UIViewController Lifecycle

    override func viewDidLoad() {
        super.viewDidLoad()
        setupAndLoadAd()
    }
    
    func setupAndLoadAd() {
        adUnitIdLabel.text = adConfigId
        adSizeLabel.text = Int(adSize.width).description + "x" + Int(adSize.height).description
        
        rewardedAd = LimelightRewardedAd(adUnitId: adConfigId)
        rewardedAd?.delegate = self
        rewardedAd?.supportSKOverlay = supportSKOverlay
        rewardedAd?.adSize = adSize
        rewardedAd?.adFormats = adFormats
        rewardedAd?.loadAd()
    }
    
    func showAd() {
        rewardedAd?.show(from: self)
    }
    
    func resetEvents() {
        onAdLoadedLabel.isEnabled = false
        onAdFailedToLoadLabel.isEnabled = false
        onAdOpenedLabel.isEnabled = false
        onAdImpressionLabel.isEnabled = false
        onAdClickedLabel.isEnabled = false
        onAdClosedLabel.isEnabled = false
        onAdRewardedLabel.isEnabled = false
    }
    
    func refreshAd() {
        resetEvents()
        refreshButton.isEnabled = false
        rewardedAd?.loadAd()
    }
}

// MARK: - LimelightRewardedAdDelegate
extension RewardedAdVC: LimelightRewardedAdDelegate {
    func onAdLoaded(rewardedAd: LimelightRewardedAd, ad: LimelightAd) {
        resetEvents()
        onAdLoadedLabel.isEnabled = true
        showAdButton.isEnabled = true
        refreshButton.isEnabled = true
    }
    
    func onAdFailedToLoad(rewardedAd: LimelightRewardedAd, error: LimelightError?) {
        resetEvents()
        onAdFailedToLoadLabel.isEnabled = true
        refreshButton.isEnabled = true
    }
    
    func onAdOpened(rewardedAd: LimelightRewardedAd) {
        onAdOpenedLabel.isEnabled = true
    }
    
    func onAdImpression(rewardedAd: LimelightRewardedAd) {
        onAdImpressionLabel.isEnabled = true
    }
    
    func onAdClicked(rewardedAd: LimelightRewardedAd) {
        onAdClickedLabel.isEnabled = true
    }
    
    func onAdClosed(rewardedAd: LimelightRewardedAd) {
        onAdClosedLabel.isEnabled = true
    }
    
    func onAdRewarded(rewardedAd: LimelightRewardedAd, reward: LimelightReward) {
        onAdRewardedLabel.isEnabled = true
    }
}
